util.go

```diff
@@ -115,3 +115,42 @@ func Concat(args ...interface{}) string {
 	}
 	return b.String()
 }
+
+func parseChainType(s string) (BaseChainType, error) {
```

Comment:
I wouldn't bother with these methods; just cast the strings to the types directly and don't worry about whether they're valid.